File: Restaurante.Repositorio/Services/Pedido/Models/PedidoModel.cs
using Restaurante.Repositorio.Services.Produto.Models;

namespace Restaurante.Repositorio.Services.Pedido.Models
{
    public class PedidoModel
    {
        public int ComandaId { get; set; }
        public int PedidoId { get; set; }
        public ProdutoListagemModel Produto { get; set; }
        public int Quantidade { get; set; }
        public string Status { get; set; }
    }
}

File: frontend/src/app/shared/models/pedido.model.ts
import { StatusPedido } from "src/app/consts/status-pedido.enum";
import { ProdutoModel } from "./produto.model";

export interface PedidoModel {

    pedidoId: number;
    produto: string;
    quantidade: number;
    valor?: number;
    dataHora?: Date;
    status: number;
    actions?: boolean;
}